### PR TITLE
Fixed various compiler warnings

### PR DESCRIPTION
**Pull request by paulhargrave**
### Type of change
- [ ] Breaking change
- [X] Non-breaking change
- [ ] Cosmetic change

### Description
Jira link: [Jira title](https://jira.tools.tax.service.gov.uk/browse/ARSSTB-151)

Fixed various compiler warnings
### Attachments or Screenshots
_Add any screenshots (if relevant) here_

### CODE REVIEW

File: app/controllers/common/FileUploadHelper.scala [hunk 1]
```diff
@@ -85,6 +86,12 @@ case class FileUploadHelper @Inject() (
             }
         case file: UploadedFile.Success   =>
           removeFile(mode, draftId, file.fileUrl.get, isLetterOfAuthority)
+        case _: Failure                   =>
+          Future.failed(
+            new RuntimeException(
+              "Shouldn't reach this point. Failure state only generated by Duplicate filenames which is handled elsewhere"
+            )
+          )
```
Comment (paulhargrave) [inline app/controllers/common/FileUploadHelper.scala:94]: This one's a bit annoying. The case type UploadedFile comtains Initiated, Success and Failure. But failure is only set after coming back to the Upload page and is only ever used to reject duplicate filenames. That happens after this point.

File: app/forms/TraderEoriNumberFormProvider.scala [hunk 2]
```diff
@@ -50,7 +50,5 @@ class TraderEoriNumberFormProvider @Inject() extends Mappings {
             Invalid(ValidationError("provideTraderEori.error.specialCharacters"))
           case _                                 => Invalid(ValidationError("provideTraderEori.error.default"))
         }
-      case _ =>
-        Invalid(ValidationError("provideTraderEori.error.default"))
```
Comment (paulhargrave) [inline app/forms/TraderEoriNumberFormProvider.scala:54]: Unreachable

File: app/navigation/CheckModeNavigator.scala [hunk 3]
```diff
@@ -84,15 +84,6 @@ object CheckModeNavigator {
...
 
-  private def whatIsYourRoleAsImporter(userAnswers: UserAnswers): Call =
-    userAnswers.get(WhatIsYourRoleAsImporterPage) match {
-      case None                                              =>
-        WhatIsYourRoleAsImporterController.onPageLoad(CheckMode, userAnswers.draftId)
-      case Some(WhatIsYourRoleAsImporter.AgentOnBehalfOfOrg) =>
-        AgentCompanyDetailsController.onPageLoad(CheckMode, userAnswers.draftId)
-      case Some(WhatIsYourRoleAsImporter.EmployeeOfOrg)      => navigateWithAuthUserType(userAnswers)
-    }
-
```
Comment (paulhargrave) [inline app/navigation/CheckModeNavigator.scala:95]: So don't allow check answers for what is your role anyway

File: app/controllers/common/FileUploadHelper.scala (hunk 1 above)
Comment (RobertLaverick) [reply app/controllers/common/FileUploadHelper.scala:94]: I'm not sure just throwing an exception works best here - an exception could use a more precise error message or we could rework this to return a Left with some logging of useful details.

Plus we don't know if the libraries behaviour around that might change later on.

File: app/forms/TraderEoriNumberFormProvider.scala (hunk 2 above)
Comment (aaocs) [reply app/forms/TraderEoriNumberFormProvider.scala:54]: Is it unreachable because Constraint is always a string?

**Review by RobertLaverick — COMMENTED** [review]
I don't love the failure case message. It feels like you're communicating to the reviewer rather than a developer who has to respond to a log that might appear

**Review by aaocs — DISMISSED** [review]

**Review by RobertLaverick — APPROVED** [review]

**Pull request merged**